espnet/nets/beam_search_transducer_espnet2.py

```diff
@@ -0,0 +1,653 @@
+"""Search algorithms for transducer models."""
+
+import numpy as np
+import torch
+
+from dataclasses import dataclass
+from typeguard import check_argument_types
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Union
+
+from espnet.nets.lm_interface import LMInterface
+from espnet2.asr.decoder.abs_decoder import AbsDecoder
+
```

Comment:
We could with some unification, input/output args were not the same that's why I added a new one for espnet2. I took the opportunity to modify into a class.

Reply:
Done!
@ShigekiKarita I had to modify `espnet/asr/pytorch_backend/asr.py`, is that okay? I'm asking because of #2521

Reply:
Please keep as it is for espent1. It is not related to this PR.

Reply:
Sorry, I thought it was your intention given first question...

If it's okay with you I'll keep it as if until PR is almost done and revert for espnet1 before merging. It was a bit complicated to unify `BeamSearchTransducer` and `ErrorCalculator` because I had to do it altogether alongside taking into account espnet1 and espnet2. As I refactored a lot of files at the same time, I couldn't properly group files / actions in commits so it will take me sometimes to revert without reverting almost everything. Sorry about that.

Reply:
Nevermind, I'll do another PR for beam search and error calculator in espnet1! The changes in this PR will be omitted after merging and updating master